xonsh/tools.py

```diff
@@ -954,6 +961,42 @@ def set_to_csv(x):
     return ','.join(x)
 
 
+def pathsep_to_set(x):
+    """Converts a os.pathsep separated string to a set of strings."""
+    if not x:
+        return set()
+    else:
+        return set(x.split(os.pathsep))
+
+
+def set_to_pathsep(x, sort=False):
+    """Converts a set to an os.pathsep separated string. The sort kwarg
+    specifies whether to sort the set prior to str conversion.
+    """
+    if sort:
+        x = sorted(x)
+    return os.pathsep.join(x)
+
+
+def is_string_seq(x):
+    """Tests if something is a sequence of strings"""
+    return (isinstance(x, abc.Sequence) and
+            all(isinstance(a, str) for a in x))
```

Comment:
This will return `True` for both `is_string_seq('HalloWorld')` and  `is_string_seq(['Hallo','World']')`